lib/handlebars/runtime.js

```diff
@@ -197,7 +197,12 @@ export function wrapProgram(container, i, fn, data, declaredBlockParams, blockPa
 export function resolvePartial(partial, context, options) {
   if (!partial) {
     if (options.name === '@partial-block') {
-      partial = options.data['partial-block'];
+      let data = options.data;
+      while (data['partial-block'] === noop) {
```

Comment:
This loop is a bit scary, I know. I'm open to suggestions for a base case to bail on.